docs/src/main/asciidoc/security-customization.adoc

```diff
@@ -620,6 +620,51 @@ are using an executor that is capable of propagating the identity (e.g. no `Comp
 to make sure that Quarkus can propagate it. For more information see the
 xref:context-propagation.adoc[Context Propagation Guide].
 
+[[observe-security-events]]
+== Observe security events
+
+Quarkus fires security events, such as authentication or authorization failure, with the `io.quarkus.security.spi.runtime.SecurityEventProducer` CDI bean.
```

Comment:
So if I create a bean of type `SecurityEventProducer`, mark it as `@Alternative` and enabled for application (`@Priority(1)`), it will be called? I didn't inspect the code too deeply, but I didn't get that impression.

I commend your desire to design an extensible API, but the nice thing about CDI events is that they already provide all the extensibility you need. If someone wants to use this API to write security events to some form of an audit log, for example, they don't need the `SecurityEventProducer` SPI -- they just need to observe those events.

Reply:
it is default bean, all you need to do is to create bean. FYI I have tests for everything :-)

> I commend your desire to design an extensible API, but the nice thing about CDI events is that they already provide all the extensibility you need. If someone wants to use this API to write security events to some form of an audit log, for example, they don't need the SecurityEventProducer SPI -- they just need to observe those events.

I'm getting that impression from all that you write. @mkouba just spent many minutes now answering my questions on Zulip. Let me comment on this PR (not just in thread).